dask_image/ndmeasure/__init__.py

```diff
@@ -97,20 +97,41 @@ def extrema(input, labels=None, index=None):
         input, labels, index
     )
 
-    min_lbl = minimum(
-        input, labels, index
-    )
-    max_lbl = maximum(
-        input, labels, index
+    out_dtype = numpy.dtype([
+        ("min_val", input.dtype),
+        ("min_pos", numpy.intp),
+        ("max_val", input.dtype),
+        ("max_pos", numpy.intp)
+    ])
+    default = numpy.zeros((), out_dtype)[()]
+
+    extrema_lbl = labeled_comprehension(
+        input, labels, index,
+        _utils._extrema, out_dtype, default, pass_positions=True
     )
-    min_pos_lbl = minimum_position(
-        input, labels, index
-    )
-    max_pos_lbl = maximum_position(
-        input, labels, index
+
+    extrema_lbl = {k: extrema_lbl[k] for k in extrema_lbl.dtype.names}
+
+    for pos_key in ["min_pos", "max_pos"]:
+        pos_1d = extrema_lbl[pos_key]
+        if not pos_1d.ndim:
+            pos_1d = pos_1d[None]
+
+        pos_nd = _utils._unravel_index(pos_1d, input.shape)
```

Comment:
Note to self: This is essentially where the tests are failing. It occurs when indices has a N-D shape with `N > 1`. Not totally sure why that affects us here and not in `maximum_position` or `minimum_position`. It may have something to do with using a structured array type to return the result. Requires further inspection.

Reply:
Addressed by PR ( https://github.com/dask/dask-image/pull/65 ) and PR ( https://github.com/dask/dask-image/pull/66 ).